File: wrappers/java/src/test/java/org/hyperledger/indy/sdk/payment/BuildMintRequestTest.java
package org.hyperledger.indy.sdk.payment;

import org.hyperledger.indy.sdk.InvalidStructureException;
import org.hyperledger.indy.sdk.payments.IncompatiblePaymentException;
import org.hyperledger.indy.sdk.payments.Payments;
import org.hyperledger.indy.sdk.payments.UnknownPaymentMethodException;
import org.junit.Test;

import java.util.concurrent.ExecutionException;

import static org.hamcrest.CoreMatchers.isA;

public class BuildMintRequestTest extends PaymentIntegrationTest {

	@Test
	public void testBuildMintRequestWorksForUnknownPaymentMethod() throws Exception {
		thrown.expect(ExecutionException.class);
		thrown.expectCause(isA(UnknownPaymentMethodException.class));

		Payments.buildMintRequest(wallet, DID_TRUSTEE, outputs, null).get();
	}

	@Test
	public void testBuildMintRequestWorksForEmptyOutputs() throws Exception {
		thrown.expect(ExecutionException.class);
		thrown.expectCause(isA(InvalidStructureException.class));

		Payments.buildMintRequest(wallet, DID_TRUSTEE, emptyArray, null).get();
	}

	@Test
	public void testBuildMintRequestWorksForIncompatiblePaymentMethods() throws Exception {
		thrown.expect(ExecutionException.class);
		thrown.expectCause(isA(IncompatiblePaymentException.class));

		Payments.buildMintRequest(wallet, DID_TRUSTEE, incompatibleOutputs, null).get();
	}
}

File: wrappers/dotnet/indy-sdk-dotnet-test/PaymentTests/BuildMintRequestTest.cs
using Hyperledger.Indy.PaymentsApi;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Threading.Tasks;

namespace Hyperledger.Indy.Test.PairwiseTests
{
    [TestClass]
    public class BuildMintRequestTest : PaymentIntegrationTestBase
    {
        [TestMethod]
        public async Task TestBuildMintRequestWorksForUnknownPaymentMethod()
        {
            var ex = await Assert.ThrowsExceptionAsync<UnknownPaymentMethodException>(() =>
                 Payments.BuildMintRequestAsync(wallet, DID_TRUSTEE, outputs, null)
             );
        }

        [TestMethod]
        public async Task TestBuildMintRequestWorksForEmptyOutputs()
        {
            var ex = await Assert.ThrowsExceptionAsync<InvalidStructureException>(() =>
                 Payments.BuildMintRequestAsync(wallet, DID_TRUSTEE, emptyArray, null)
             );
        }

        [TestMethod]
        public async Task TestBuildMintRequestWorksForIncompatiblePaymentMethods()
        {
            var ex = await Assert.ThrowsExceptionAsync<IncompatiblePaymentMethodsException>(() =>
                 Payments.BuildMintRequestAsync(wallet, DID_TRUSTEE, incompatibleOutputs, null)
             );
        }
    }
}
